### PR TITLE
Link binaryen tools against the dylib

### PR DESCRIPTION
**Pull request by dschuff**
When building the libbinaryen dynamic library, also link the binaryen tools against it. This reduces combined tool size on mac from 76M to 2.8M

### CODE REVIEW

**Comment by dschuff** [comment]
Is there no longer a way to run CI against a branch without opening a PR?

**Comment by dschuff** [comment]
Also, I don't understand why this build was cancelled?

**Comment by dschuff** [comment]
@sbc100 any ideas?

**Comment by sbc100** [comment]
> Is there no longer a way to run CI against a branch without opening a PR?

No.  The reason is that if we do that then we when a normal developer opens a PR it causes the branch builder and the PR builder to trigger, needlessly doubling the work.

Unless you can think of a better way, the current solution is to open a PR with WIP in the title, or just don't assign any reviewers.

**Comment by sbc100** [comment]
I think if any job fails it cancels all the other ones.

File: CMakeLists.txt [hunk 1]
```diff
@@ -280,95 +280,26 @@ install(TARGETS binaryen
...
+binaryen_add_executable(wasm-shell src/tools/wasm-shell.cpp)
+binaryen_add_executable(wasm-metadce src/tools/wasm-metadce.cpp)
+binaryen_add_executable(asm2wasm src/tools/asm2wasm.cpp)
+binaryen_add_executable(wasm2js src/tools/wasm2js.cpp)
+binaryen_add_executable(wasm-emscripten-finalize src/tools/wasm-emscripten-finalize.cpp)
+binaryen_add_executable(wasm-as src/tools/wasm-as.cpp)
+binaryen_add_executable(wasm-dis src/tools/wasm-dis.cpp)
+binaryen_add_executable(wasm-ctor-eval src/tools/wasm-ctor-eval.cpp)
+binaryen_add_executable(wasm-reduce src/tools/wasm-reduce.cpp)
+
```
Comment (sbc100) [inline CMakeLists.txt:302]: Perhaps land this refactoring change first, and then try to land the dylib change as a followup?

Comment (sbc100) [reply CMakeLists.txt:302]: Are we doing to need to do horrible `-rpath` hacks though?

Comment (dschuff) [reply CMakeLists.txt:302]: I don't think so? It seems to work on my machine. CMake does do some kind of rpath magic automatically.

**Comment by dschuff** [comment]
> I think if any job fails it cancels all the other ones.

sort of makes sense, that behavior is kind of annoying when there's a known failure (like right now on master) but you want to see the other tests.

**Comment by sbc100** [comment]
> > I think if any job fails it cancels all the other ones.
> 
> sort of makes sense, that behavior is kind of annoying when there's a known failure (like right now on master) but you want to see the other tests.

It could be configurable in github actions.. I've just not seen the option for it yet.

**Comment by dschuff** [comment]
OK, I think this should work now. I had to disable the shared build on windows for now because we don't dllexport anything, so we can't actually link the tools against the dylib yet.
We should talk about whether it's worthwhile to do this. The size win is pretty substantial (at least unzipped, I haven't checked zipped yet).

**Comment by dschuff** [comment]
Going to go ahead and land this. I thought about pulling out the cmake refactor but it didn't seem worthwhile.

**Pull request merged**